src/Features/CSharp/Portable/CodeLens/CSharpCodeLensDisplayInfoService.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeLens
+{
+    [ExportLanguageService(typeof(ICodeLensDisplayInfoService), LanguageNames.CSharp), Shared]
+    internal sealed class CSharpCodeLensDisplayInfoService : ICodeLensDisplayInfoService
+    {
+        private static readonly SymbolDisplayFormat DefaultFormat =
+            SymbolDisplayFormat.CSharpErrorMessageFormat.RemoveMemberOptions(
+                SymbolDisplayMemberOptions.IncludeExplicitInterface);
+
+        // Matches default ToDisplayString except removing global namspace and namespaces
+        private static readonly SymbolDisplayFormat ShortFormat =
+            DefaultFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted)
+                .WithTypeQualificationStyle(SymbolDisplayTypeQualificationStyle.NameAndContainingTypes);
+
+        /// <summary>
+        /// Returns the node that should be displayed
+        /// </summary>
+        public SyntaxNode GetDisplayNode(SyntaxNode node)
+        {
+            while (true)
+            {
+                switch (node.Kind())
+                {
+                    // LocalDeclarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.LocalDeclarationStatement:
+                        var localDeclarationNode = (LocalDeclarationStatementSyntax) node;
+                        node = localDeclarationNode.Declaration.Variables.First();
+                        continue;
+
+                        // Field and event declarations do not have symbols themselves, you need a variable declarator
+                    case SyntaxKind.FieldDeclaration:
+                    case SyntaxKind.EventFieldDeclaration:
+                        var fieldNode = (BaseFieldDeclarationSyntax) node;
+                        node = fieldNode.Declaration.Variables.First();
+                        continue;
+
+                        // Variable is a field without access modifier. Parent is FieldDeclaration
+                    case SyntaxKind.VariableDeclaration:
+                        node = node.Parent;
+                        continue;
```

Comment:
Yes.

---
In reply to: [74126795](https://github.com/dotnet/roslyn/pull/13037#discussion_r74126795) [](ancestors = 74126795)